### ISSUE TITLE
[Issue]: Displaying actor photos with local metadata

### ISSUE THREAD

**Issue by deltonio2** [jellyfin/jellyfin#8103]
### Please describe your bug

When you install a fresh new Jellyfin server, define all the metadata to be stored locally and create a Jellyfin library based on an existing folder with multiple files, the picture of the actors doesn't show in the interface, even when the scan is 100% finish.

I tried with a Debian server and a Synology server on Docker, on the web interface of Jellyfin or the Android TV app, the problem is the same.

However, if I click on the actor, his photo is displayed correctly.
If I go back to the film, the thumbnails display well, and will also be on next connections:

![image](https://user-images.githubusercontent.com/8058756/178256531-6cf70c8c-d0f1-48bd-88d4-7e5586580177.png)

![image](https://user-images.githubusercontent.com/8058756/178256592-30b9d46f-352b-4398-b57a-b1a3a0304577.png)

![image](https://user-images.githubusercontent.com/8058756/178256673-347f377e-729e-4e0d-9b9e-26f23c9a9a64.png)

When I add a new movie in the library (after that Jellyfin was installed), the problem doesn't occur. All the actors that play in the new movies have their profile picture correctly loaded without having to click on it.

### Jellyfin Version

Other

### if other:

10.8.1

### Environment

```markdown
- OS:Debian and DSM
- Virtualization:N/A and Docker
- Clients:Android TV
- Browser:Google Chrome 103
- Hardware Acceleration:None
- Plugins:None
- Reverse Proxy:None
- Networking:Host
- Storage:local
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by RainyDAy59** [comment]
I'm experiencing this too using Jellyfin 10.8.4.  Only when I click on the actor box will their image appear.

**Comment by gyre-ri** [comment]
Exactly the same for me...

**Comment by bhaveman** [comment]
Same in 10.8.5

**Comment by j-insan3** [comment]
Same issue inn 10.8.7

**Comment by deltonio2** [comment]
The issue is still there in version 10.8.9

**Comment by deltonio2** [comment]
I believe I understood what causes the problem, when Jellyfin detects a new movie in the directory which has its metadata already downloaded (NFO and JPG file), it will add it to the library but it will not download the photos of the actors who star in the film.

I'm doing an rsync of the main server directory where my movies are on a secondary server. On my main server, I see the photos of the actors for the new movies that I add. On the secondary server, even the new movies don't have the cast photos added.

**Comment by gummykage** [comment]
Just had a RAID fail so I had to setup a new server... Problem still persists as of 10.8.10

**Comment by deltonio2** [comment]
I found a workaround that allows you to retrieve all the photos of the actors at a given time.

I don't understand at all how Jellyfin works in backend but I know that by clicking on an actor, it loads his photo that remains for the future. So I do as if I was clicking on all the actors present in the Jellyfin server. This is definitely a bad solution that is completely unoptimized, but it works.

Here is a PowerShell line that does the job:

`(Invoke-RestMethod -Uri "http://youJellyfinServer:8096/emby/Persons?api_key=yourAPIKey").Items | ForEach-Object { Invoke-RestMethod -Uri "http://youJellyfinServer:8096/Users/yourUserID/Items/$($_.Id)?api_key=yourAPIKey" }`

Things you need to change to make it works on your server:

- **youJellyfinServer:** the hostname of your Jellyfin server
- **yourAPIKey:** can be retrieved by going to the Dashboard, API Keys, + button, enter any name, get the generated API Key
- **yourUserID:** can be retrieved by going to the Dashboard, Users, click on your (or another) user, get what is in the URL after "userId="

**Comment by ReaderGuy42** [comment]
@deltonio2 How would I do this on Linux or rather Raspberry Pi? Your powershell script doesn't work there, and I can't get powershell to install on it. Specifically it's complaining about `.Items` I think.

Thanks!

**Comment by deltonio2** [comment]
@ReaderGuy42 you can run this script from any station that has access to your server, If you have a Windows PC on hand, PowerShell is installed by default, otherwise you can install PowerShell Core on macOS or Linux.

You will find the Microsoft help page for installing PowerShell Core on Raspbian [here](https://learn.microsoft.com/en-us/powershell/scripting/install/community-support?view=powershell-7.3#raspberry-pi-os).

I tested my script on default Windows PowerShell version 5.1.19041 and on PowerShell Core last version 7.3.7 and it works on both.

**Comment by ReaderGuy42** [comment]
Thanks that worked! However, now I'd like to schedule that as a task, as you mention in your earlier comment. But I can't figure out how to do that. Any advice? Thanks!

**Comment by deltonio2** [comment]
On Windows you can do it with the scheduled tasks tool, on Raspbbery/Linux you can do it with cron.

You will easily find examples by searching on the internet, e.g. [here](https://www.sharepointdiary.com/2013/03/create-scheduled-task-for-powershell-script.html) for Windows and [here](https://www.lewisroberts.com/2018/02/07/powershell-core-6-raspberry-pi/) for Raspbian.

**Comment by ReaderGuy42** [comment]
Great, another question: The second script in your comment that's meant to only find images that are missing, but that seems to not be working for me. It just outputs the same actor name over and over again.

```
{@{Name=Adam Bernstein; ServerId=ce8e839cc60247e2b7bffa9231e7a72f; Id=1bdaaa35…
{@{Name=Adam Bernstein; ServerId=ce8e839cc60247e2b7bffa9231e7a72f; Id=1bdaaa35…
{@{Name=Adam Bernstein; ServerId=ce8e839cc60247e2b7bffa9231e7a72f; Id=1bdaaa35…
```
the first script worked though. But for the cronjob I'd like it to only find the missing images not replace them all.

**Comment by deltonio2** [comment]
Weird, I don't understand why the same actor comes over and over again.
Try this command which removes the duplicates:

`(Invoke-RestMethod -Uri "http://youJellyfinServer:8096/emby/Persons?api_key=yourAPIKey").Items | Where-Object { [string]::IsNullOrEmpty($_.ImageTags) } | Sort-Object -Property Id -Unique | ForEach-Object { Invoke-RestMethod -Uri "http://youJellyfinServer:8096/Users/yourUserID/Items/$($_.Id)?api_key=yourAPIKey" }`

Please note that the commands I provide check all the actors who do not have a photo, but do not differentiate between an actor who does not yet have his photo on your server and an actor who does not have a photo at all in the website you rely on for your metadata (e.g. TMDB).

**Comment by ReaderGuy42** [comment]
That seems to have worked! Thank you :)

**Comment by deltonio2** [comment]
@felix920506 you moved this ticket to "Insufficient Data : Additional information is required".

What additional information is needed?

**Comment by merlink01** [comment]
Same Error on 10.8.11, can I provide some data to help???

**Comment by felix920506** [comment]
@deltonio2 It was moved to Insufficient Data because it was unclear what is actually going on here and/or if it was actually a Jellyfin issue when I was skimming through this ticket. The Insufficient Data tag is meant to be like a dumping ground sort of things that needs some investigating.

@merlink01 Yes you absolutely can provide data to help.

--

We would like the following:
- Jellyfin logs (can be accessed in the dashboard)
- Test results on the master branch (if possible)

**Comment by sevenrats** [comment]
@deltonio2  @merlink01 
These aren't fetched during the update people task?
Try manually running the update people task and see if this works instead.

**Comment by deltonio2** [comment]
@sevenrats No they are not. I tried running the task manually again, it runs fine but does not download the missing photos from the server.

@felix920506 I think the problem is linked to Jellyfin because it's 100% reproducible, I have done numerous tests on several different servers with different OS in 100% of cases the problem arises when the metadata is already present stored locally with the films.

As I had put above https://github.com/jellyfin/jellyfin/issues/8103#issuecomment-1730783019, it is possible to recover all the actors' photos working with the API but it's only a workaround which takes a long time to execute and even doing a scheduled task, the problem will always be present between adding a new movie and running this script.

Here is the export of the last log file just after I manually launched the actor refresh task: [Log.txt](https://github.com/jellyfin/jellyfin/files/12904243/Log.txt)

For the test results on the master branch I'm not sure how to do this?

If you need other logs or anything else I can provide it to you.

**Comment by felix920506** [comment]
@deltonio2 install a dotnet runtime, clone the repo and do `dotnet run Jellyfin.Server` in the directory

**Comment by sevenrats** [comment]
@deltonio2 i might not do this because of the risk of overwriting your database.  downgrades from unstable are not supported, so if you take this route make sure you have backed up your working installation.

**Comment by sevenrats** [comment]
also, there is no need because I have confirmed that this happens in latest unstable as well.

**Comment by nbnrvtn** [comment]
Hi, is there any plan to resolve this?

**Comment by merlink01** [comment]
For those of you who need a crossplatform solution, here is a python script:

Attention, it coultd take long depending on the size of your library.

I speeded it up using 5 threads in parrallel.

```
import requests
import threading
import time

apikey = "your_api_token"

us = requests.get(f"http://yellyfin_ip:jellyfin_port/emby/Users?api_key={apikey}").json()
uids = []
for u in us:
    uids.append(u["Id"])

url = f"http://yellyfin_ip:jellyfin_port/emby/Persons?api_key={apikey}&enableImages=false"
ps = requests.get(url).json()["Items"]


def update_pid(pid):
    for userid in uids:
        url = f"http://yellyfin_ip:jellyfin_port/Users/{userid}/Items/{pid}?api_key={apikey}"
        requests.get(url).json()["Name"]


pslen = len(ps)
count = 0
maxthreads = 5
for p in ps:
    count += 1

    while 1:
        tcount = 0
        for t in threading.enumerate():
            if t.name == "worker":
                tcount += 1

        if tcount < maxthreads:
            break

    print(f"{count}/{pslen}", p["Name"])
    time.sleep(0.1)

    pid = p["Id"]
    t = threading.Thread(target=update_pid, args=(pid,), name="worker")
    t.start()

```

**Comment by thermionic** [comment]
I also hit this on 10.8.13, and have confirmed on 10.9 and 10.9.1

**Comment by Shadowghost** [comment]
It was never implemente, so this is more of a feature request than a bug.
Actors (or any kind of person) currently only have one image. If we'd want to implement it, we'd need one global fallback image and additional images which take preference if they are in a movie/series folder - this is currently not supported by our data layout.

**Comment by bhaveman** [comment]
I may be misunderstanding your response, but the original issue isn't about multiple pictures for the same actor; rather it's that the one picture of an actor does not display until the actor has been selected, at which point it continues to display properly.

Almost like the request to get the one image doesn't happen until the actor is selected/clicked.

**Comment by thermionic** [comment]
@Shadowghost the issue that I saw was that the Actor image was not grabbed until the actor "page" was accessed, I confirmed this by checking to see if it existed in the metadata/People directory structure before and after the actor page was accessed.

**Comment by felix920506** [comment]
This probably needs a better name to avoid confusion in the future.

**Comment by thermionic** [comment]
[withmetadata.log](https://github.com/jellyfin/jellyfin/files/15312659/withmetadata.log)
[withoutmetadata.log](https://github.com/jellyfin/jellyfin/files/15312660/withoutmetadata.log)

@Shadowghost 

Basic deployment of 10.9.1 (docker on Ubuntu 22.04) ) and a single very small test library.

Initial run was against library with metadata already populated, and one can see the multiple "Cannot get image dimensions" for the actors, and the actors do not show until accessed (at which point they load), interestingly Directors were shown

For the second run, the config directory was emptied (so no possibility of anything being carried over) and all metadata (created with MediaElch) was removed from the directory structure, as the log shows there were no "Cannot get image dimensions" from the log and actor image were shown against the films.

logs captured with `docker logs jellyfin > filename.log`

**Comment by thermionic** [comment]
[debugwithmetadata.log](https://github.com/jellyfin/jellyfin/files/15313574/debugwithmetadata.log)

As I have been cleaning the directory between launches, the attached log is slightly different as initial account creation was done, then shut down and added logging.json configured for "debug", then restarted and library added.

This time I managed to see the "File not found" (for each actor), and the behaviour was as before with only Directors shown, still able to click through to the Actor which loaded the actor image, and the log shows the download of the actor image.

**Comment by ouafouaf** [comment]
I face the same issue with actors thumbs not working when using local metadata from nfo files (created with tinymediamanager in my case). I couldn't find a way to fix it in jellyfin or in tinymediamanager. Actor image is only fetched and loaded in jellyfin after opening the actors page. The powershell script provided by [deltonio2](https://github.com/deltonio2) is a nice workaround, but it's a workaround nonetheless.

Is there a feature request on this issue?

**Comment by vonwanja** [comment]
Same Issue here ...

**Comment by kuddukan1** [comment]
This is still a problem.

**Comment by luke-liuqi** [comment]
This is still a problem in 10.10.0

**Comment by Alethio36** [comment]
I can confirm this is still a problem. Currently running the python script. Dude was not kidding, it's gonna take two days to finish

**Comment by soakes** [comment]
> I can confirm this is still a problem. Currently running the python script. Dude was not kidding, it's gonna take two days to finish

Try the powershell script.  Its much quicker.  If you dont use Windows, just use docker.

**Comment by Alethio36** [comment]
i just felt committed to what i chose. the powershell options took at least a hour before i gave up. for future reference my library is about 2.5tb of movies and shows. im not really sure whats going on with this issue, or why a rescan of people doesnt solve the issue. i am in the process of moving to a new JF instance (new architecture, moving from truenas scale to a proper docker setup). i moved my media library but made a brand new JF instance. 

if i can provide any information to help solve this i would love to

**Comment by soakes** [comment]
> i just felt committed to what i chose. the powershell options took at least a hour before i gave up. for future reference my library is about 2.5tb of movies and shows. im not really sure whats going on with this issue, or why a rescan of people doesnt solve the issue. i am in the process of moving to a new JF instance (new architecture, moving from truenas scale to a proper docker setup). i moved my media library but made a brand new JF instance.
> 
> if i can provide any information to help solve this i would love to

The PowerShell method is quite simple, first go into JF `users` section, select any user i.e admin (or create a new one).  Now look at the URL in your URL bar, it will be something like `https://example.io/web/#/dashboard/users/profile?userId=302ecaf45dcb48c48fbd1935ed02aded` - you want to make a note of the userID part after the equils sign.

You will now need a new API key, so go create one.  For example, 
`17c8381460bb4c17bf42779d51cd6d53	demo	26/07/2024 09:10`  

Next adjust the following snipet of code with the info you obtained above.

```
(Invoke-RestMethod -Uri "https://example.io/emby/Persons?api_key=17c8381460bb4c17bf42779d51cd6d53").Items | Where-Object { [string]::IsNullOrEmpty($_.ImageTags) } | ForEach-Object { Invoke-RestMethod -Uri "https://example.io/Users/302ecaf45dcb48c48fbd1935ed02aded/Items/$($_.Id)?api_key=17c8381460bb4c17bf42779d51cd6d53" }
```

As you can see the API key is used in two places, the same with the URL, but only apears once for the `userID`. 

Run this code you crafted in the powershell. If no native powershell, use docker version, works fine as thats what I do.

This will fix any missing artwork, it takes a while but not too long and far quicker then the python method.

HTH

**Comment by Alethio36** [comment]
Is there some way to indicate progress with the powershell method? It doesn't seem to go in alphabetical order or in any order I can see. It would be nice to have some sort of estimate on how long things will take.

**Comment by soakes** [comment]
> Is there some way to indicate progress with the powershell method? It doesn't seem to go in alphabetical order or in any order I can see. It would be nice to have some sort of estimate on how long things will take.

Not that im aware of away, remmeber its just running over a list of looped records.  

I am also not the original author, the guy who created the PS script is called @deltonio2, he is the one to thank, I just mentioned how its done as some people like yourself didnt understand his instructions.  

Just leave it running, it be done in less then an hour.

**Comment by Alethio36** [comment]
i appreciate and thank you sir

**Comment by deltonio2** [comment]
I see @soakes script just as I'm finishing mine :-)
Here it is anyway, perhaps the progress is more visible:
```
Write-Host "Please wait, the progress bar will appear soon."
$actors = (Invoke-RestMethod -Uri "http://youJellyfinServer:8096/emby/Persons?api_key=yourAPIKey").Items | Where-Object { [string]::IsNullOrEmpty($_.ImageTags) } | Sort-Object -Property Id -Unique
$numberOfActors = $actors.Count
$count=0
Clear-Host
$actors | ForEach-Object {
	$percentComplete = $count / $numberOfActors * 100
	$roundedPercent = [math]::Round($percentComplete)
	Write-Progress -Activity "Downloading actors photos in progress" -Status "$count actors completed out of $numberOfActors. $roundedPercent% complete:" -PercentComplete $percentComplete
	$null = Invoke-RestMethod -Uri "http://youJellyfinServer:8096/Users/yourUserID/Items/$($_.Id)?api_key=yourAPIKey"
	$count++
}
```
And reading the other comments, I recall that PowerShell has been cross-platform for several years now.
Microsoft documentation for PowerShell exists for [macOS](https://learn.microsoft.com/en-us/powershell/scripting/install/installing-powershell-on-macos?view=powershell-7.4), [Debian](https://learn.microsoft.com/en-us/powershell/scripting/install/install-debian?view=powershell-7.4), [Ubuntu](https://learn.microsoft.com/en-us/powershell/scripting/install/install-ubuntu?view=powershell-7.4) and others.

**Comment by soakes** [comment]
> I see @soakes script just as I'm finishing mine :-) Here it is anyway, perhaps the progress is more visible:
> 
> ```
> Write-Host "Please wait, the progress bar will appear soon."
> $actors = (Invoke-RestMethod -Uri "http://youJellyfinServer:8096/emby/Persons?api_key=yourAPIKey").Items | Where-Object { [string]::IsNullOrEmpty($_.ImageTags) } | Sort-Object -Property Id -Unique
> $numberOfActors = $actors.Count
> $count=0
> Clear-Host
> $actors | ForEach-Object {
> 	$percentComplete = $count / $numberOfActors * 100
> 	$roundedPercent = [math]::Round($percentComplete)
> 	Write-Progress -Activity "Downloading actors photos in progress" -Status "$count actors completed out of $numberOfActors. $roundedPercent% complete:" -PercentComplete $percentComplete
> 	$null = Invoke-RestMethod -Uri "http://youJellyfinServer:8096/Users/yourUserID/Items/$($_.Id)?api_key=yourAPIKey"
> 	$count++
> }
> ```
> 
> And reading the other comments, I recall that PowerShell has been cross-platform for several years now. Microsoft documentation for PowerShell exists for [macOS](https://learn.microsoft.com/en-us/powershell/scripting/install/installing-powershell-on-macos?view=powershell-7.4), [Debian](https://learn.microsoft.com/en-us/powershell/scripting/install/install-debian?view=powershell-7.4), [Ubuntu](https://learn.microsoft.com/en-us/powershell/scripting/install/install-ubuntu?view=powershell-7.4) and others.

The adjustments which I made @deltonio2 was using ChatGPT as I don't know PS too well.  I am a Linux man, so bash is my main fortee.  I still say yours is probbaly better :) its human made afterall, us humans need to stay together lol :).

**Comment by PennyWafel** [comment]
As of december 2nd of 2024, this issue remains. None of my actor pictures are showing up until I click them.

**Comment by sevenrats** [comment]
Thanks for contributing. The traditional way to plus one a gh issue is to thumbs up the op. This issue has already been flagged confirmed, so no additional reporting is necessary to confirm it.

**Comment by pablobass** [comment]
> I adapted this script for bash [here](https://github.com/nagug/Jellyfin-Actor-Refresh-Script) with some visible progress and feedback.

Thanks! It worked perfectly.

Some clarifications in case someone else encounters the same situation.

On one hand, my server has ICMP blocked, and the script checks if it is online via ping, so in that case, it exits with the message "server not found" as soon as you run it. If that's the case, look for the lines that perform the ICMP check, comment them out, and that's it.

On the other hand, I also have it under a domain with https and access only from the base domain (i.e., I don't access it with ip:port). In that case, complete the api_url variable with "https://example.com" and it also works perfectly.

Thanks again for this.

I hope they can resolve the problem soon so we don't have to keep applying workarounds.

**Comment by nagug** [comment]
yes, that should work.

**Comment by SwaggyMacro** [comment]
> I adapted this script for bash [here](https://github.com/nagug/Jellyfin-Actor-Refresh-Script) with some visible progress and feedback.

First of all, thank you! However, I encountered this error and had no idea how to resolve it.
![Image](https://github.com/user-attachments/assets/88c26f39-3aea-494f-a535-be5db3e50acc)
So I created this [jellyfin-actor-processor](https://github.com/SwaggyMacro/jellyfin-actor-processor), It works!

**Comment by nosebeggar** [comment]
> I found a workaround that allows you to retrieve all the photos of the actors at a given time.
> 
> I don't understand at all how Jellyfin works in backend but I know that by clicking on an actor, it loads his photo that remains for the future. So I do as if I was clicking on all the actors present in the Jellyfin server. This is definitely a bad solution that is completely unoptimized, but it works.
> 
> Here is a PowerShell line that does the job:
> 
> `(Invoke-RestMethod -Uri "http://youJellyfinServer:8096/emby/Persons?api_key=yourAPIKey").Items | ForEach-Object { Invoke-RestMethod -Uri "http://youJellyfinServer:8096/Users/yourUserID/Items/$($_.Id)?api_key=yourAPIKey" }`
> 
> Things you need to change to make it works on your server:
> 
>     * **youJellyfinServer:** the hostname of your Jellyfin server
> 
>     * **yourAPIKey:** can be retrieved by going to the Dashboard, API Keys, + button, enter any name, get the generated API Key
> 
>     * **yourUserID:** can be retrieved by going to the Dashboard, Users, click on your (or another) user, get what is in the URL after "userId="
> 
> 
> **EDIT:**
> 
> You can use this line to only load photos of actors on your server who don't have any:
> 
> `(Invoke-RestMethod -Uri "http://youJellyfinServer:8096/emby/Persons?api_key=yourAPIKey").Items | Where-Object { [string]::IsNullOrEmpty($_.ImageTags) } | ForEach-Object { Invoke-RestMethod -Uri "http://youJellyfinServer:8096/Users/yourUserID/Items/$($_.Id)?api_key=yourAPIKey" }`
> 
> If like me you have servers whose libraries are replicated on another, you can:
> 
>     * make a scheduled task of this command to add the photos of the new actors added
> 
>     * move the metadata folder from the main server to a location replicated with the other servers, then make natural links for all servers to point from the default metadata emplacement to the new location
> 
> 
> **EDIT 2:**
> 
> This command avoids searching for the same actor several times:
> 
> `(Invoke-RestMethod -Uri "http://youJellyfinServer:8096/emby/Persons?api_key=yourAPIKey").Items | Where-Object { [string]::IsNullOrEmpty($_.ImageTags) } | Sort-Object -Property Id -Unique | ForEach-Object { Invoke-RestMethod -Uri "http://youJellyfinServer:8096/Users/yourUserID/Items/$($_.Id)?api_key=yourAPIKey" }`
> 
> Please note that the commands check all the actors who do not have a photo, but do not differentiate between an actor who does not yet have his photo on your server and an actor who does not have a photo at all in the website you rely on for your metadata (e.g. TMDB).

For anyone who happens to stumble upon this (It's been more than 2 years now, but the issue still persists), here is the linux version:
```
#!/bin/bash

API_KEY="ajksjdijaskdjaiksjdkajsd"
BASE_URL="https://jellyfin.example.com"
USER="theUsernameYouLogInWith"
#Go to https://jellyfin.example.com/web/#/dashboard/users and "rightclick > copy link" on your portrait to get your user ID
USERID="31fd53576d384b288d055549dad87a4a"

# Fetch all persons
response=$(curl -s "${BASE_URL}/emby/Persons?api_key=${API_KEY}")

# Filter persons with empty ImageTags, sort by Id, and remove duplicates
ids=$(echo "$response" | jq -r '.Items[] | select(.ImageTags == null or .ImageTags == {}) | .Id' | sort -u)

# Fetch details for each person
for id in $ids; do
    curl -s "${BASE_URL}/Users/${USERID}/Items/${id}?api_key=${API_KEY}"
done
```

You can just throw this into a cronjob every night, this forces jellyfin to redownload pictures for the actors without one.

**Comment by pablobass** [comment]
here is the linux version:
> 
> ```
> #!/bin/bash
> 
> API_KEY="ajksjdijaskdjaiksjdkajsd"
> BASE_URL="https://jellyfin.example.com"
> USER="theUsernameYouLogInWith"
> #Go to https://jellyfin.example.com/web/#/dashboard/users and "rightclick > copy link" on your portrait to get your user ID
> USERID="31fd53576d384b288d055549dad87a4a"
> 
> # Fetch all persons
> response=$(curl -s "${BASE_URL}/emby/Persons?api_key=${API_KEY}")
> 
> # Filter persons with empty ImageTags, sort by Id, and remove duplicates
> ids=$(echo "$response" | jq -r '.Items[] | select(.ImageTags == null or .ImageTags == {}) | .Id' | sort -u)
> 
> # Fetch details for each person
> for id in $ids; do
>     curl -s "${BASE_URL}/Users/${USERID}/Items/${id}?api_key=${API_KEY}"
> done
> ```
> 
> You can just throw this into a cronjob every night, this forces jellyfin to redownload pictures for the actors without one.

First of all, thanks for this!

Sadly, when I run it, it gives me this error:

jq: parse error: Invalid numeric literal at line 1, column 7

I'm using Jellyfin 10.10.5 in ubuntu 24.04

**Comment by need4swede** [comment]
To add to @nosebeggar, here's my version. It includes a little more verbose logging...

```
#!/bin/bash

API_KEY=""
BASE_URL=""
USER=""
# Go to ~/dashboard/users and "rightclick > copy link" on your portrait to get your user ID
USERID=""
# /path/to/file.log
LOG_FILE=""

# Function to log and print at the same time with timestamp
log_and_print() {
    timestamp=$(date "+%Y-%m-%d %H:%M:%S")
    echo "[$timestamp] $1"
    echo "[$timestamp] $1" >> "$LOG_FILE"
}

# Create log directory if it doesn't exist
log_and_print "Script initialization: Creating log directory if needed"
mkdir -p "$(dirname "$LOG_FILE")"
log_and_print "Log will be saved to: $LOG_FILE"

# Begin logging
log_and_print "========================================================"
log_and_print "STARTING JELLYFIN PERSON IMAGE CHECK"
log_and_print "========================================================"
log_and_print "Configuration:"
log_and_print "  - Base URL: $BASE_URL"
log_and_print "  - User: $USER"
log_and_print "  - User ID: $USERID"
log_and_print "  - API Key: ${API_KEY:0:5}... (truncated for security)"

# Fetch all persons
log_and_print "========================================================"
log_and_print "PHASE 1: FETCHING ALL PERSONS FROM JELLYFIN"
log_and_print "Making API request to: ${BASE_URL}/emby/Persons"
log_and_print "Method: GET, Timeout: Default"
start_time=$(date +%s)
response=$(curl -s "${BASE_URL}/emby/Persons?api_key=${API_KEY}")
end_time=$(date +%s)
duration=$((end_time - start_time))
log_and_print "API request completed in $duration seconds"

# Check if fetch was successful
if [ -z "$response" ]; then
    log_and_print "ERROR: Failed to fetch persons. Empty response received."
    log_and_print "Please check your API key and BASE_URL configuration."
    log_and_print "Aborting script execution."
    exit 1
fi

# Get total number of persons
total_persons=$(echo "$response" | jq '.TotalRecordCount')
log_and_print "Successfully fetched data for $total_persons total persons"

# Check if jq is working properly
if ! echo "$response" | jq '.' > /dev/null 2>&1; then
    log_and_print "ERROR: Failed to parse JSON response. Check if jq is installed and the API returned valid JSON."
    log_and_print "First 100 characters of response: ${response:0:100}..."
    log_and_print "Aborting script execution."
    exit 1
fi

# Filter persons with empty ImageTags, sort by Id, and remove duplicates
log_and_print "========================================================"
log_and_print "PHASE 2: FILTERING PERSONS WITH MISSING IMAGES"
log_and_print "Applying filter: Items with null or empty ImageTags"
log_and_print "Using jq query: '.Items[] | select(.ImageTags == null or .ImageTags == {}) | .Id'"
start_time=$(date +%s)
ids=$(echo "$response" | jq -r '.Items[] | select(.ImageTags == null or .ImageTags == {}) | .Id' | sort -u)
end_time=$(date +%s)
duration=$((end_time - start_time))
log_and_print "Filtering completed in $duration seconds"

# Log how many IDs were found
if [ -z "$ids" ]; then
    id_count=0
    log_and_print "No persons with missing images found. Script will exit."
    log_and_print "========================================================"
    log_and_print "SCRIPT COMPLETED: No processing needed"
    exit 0
else
    id_count=$(echo "$ids" | wc -l)
    log_and_print "Found $id_count persons with missing images out of $total_persons total ($((id_count*100/total_persons))%)"
fi

# Fetch details for each person
log_and_print "========================================================"
log_and_print "PHASE 3: FETCHING DETAILS FOR EACH PERSON WITH MISSING IMAGES"
log_and_print "Will process $id_count persons individually"

count=0
process_start_time=$(date +%s)
for id in $ids; do
    count=$((count+1))
    person_start_time=$(date +%s)
    log_and_print "------------------------------------------------------"
    log_and_print "PROCESSING PERSON [$count/$id_count]: ID = $id"
    log_and_print "Making API request to: ${BASE_URL}/Users/${USERID}/Items/${id}"
    
    result=$(curl -s "${BASE_URL}/Users/${USERID}/Items/${id}?api_key=${API_KEY}")
    
    # Check if result is valid
    if [ -z "$result" ]; then
        log_and_print "WARNING: Empty response received for ID: $id. Skipping."
        continue
    fi
    
    # Extract person details
    name=$(echo "$result" | jq -r '.Name // "Unknown"')
    type=$(echo "$result" | jq -r '.Type // "Unknown"')
    overview=$(echo "$result" | jq -r '.Overview // "No overview available"' | head -c 50)
    if [ ${#overview} -ge 50 ]; then
        overview="${overview}..."
    fi
    
    person_end_time=$(date +%s)
    person_duration=$((person_end_time - person_start_time))
    
    log_and_print "Person details:"
    log_and_print "  - Name: $name"
    log_and_print "  - Type: $type"
    log_and_print "  - Overview: $overview"
    log_and_print "  - Has ImageTags: $(echo "$result" | jq 'if .ImageTags and (.ImageTags | length > 0) then "Yes" else "No" end')"
    log_and_print "Processing completed in $person_duration seconds"
    
    # Progress calculation
    percent_complete=$((count*100/id_count))
    log_and_print "Overall progress: $percent_complete% complete ($count/$id_count)"
done

process_end_time=$(date +%s)
total_duration=$((process_end_time - process_start_time))
avg_time=$(echo "scale=2; $total_duration / $id_count" | bc 2>/dev/null || echo "N/A")

log_and_print "========================================================"
log_and_print "SCRIPT COMPLETED SUCCESSFULLY"
log_and_print "Summary:"
log_and_print "  - Total persons in Jellyfin: $total_persons"
log_and_print "  - Persons with missing images: $id_count"
log_and_print "  - Percentage missing: $((id_count*100/total_persons))%"
log_and_print "  - Total processing time: $total_duration seconds"
log_and_print "  - Average time per person: $avg_time seconds"
log_and_print "========================================================"
```

**Comment by HHUBSS** [comment]
> For anyone who happens to stumble upon this (It's been more than 2 years now, but the issue still persists), here is the linux version:
> 
> ```
> #!/bin/bash
> 
> API_KEY="ajksjdijaskdjaiksjdkajsd"
> BASE_URL="https://jellyfin.example.com"
> USER="theUsernameYouLogInWith"
> #Go to https://jellyfin.example.com/web/#/dashboard/users and "rightclick > copy link" on your portrait to get your user ID
> USERID="31fd53576d384b288d055549dad87a4a"
> 
> # Fetch all persons
> response=$(curl -s "${BASE_URL}/emby/Persons?api_key=${API_KEY}")
> 
> # Filter persons with empty ImageTags, sort by Id, and remove duplicates
> ids=$(echo "$response" | jq -r '.Items[] | select(.ImageTags == null or .ImageTags == {}) | .Id' | sort -u)
> 
> # Fetch details for each person
> for id in $ids; do
>     curl -s "${BASE_URL}/Users/${USERID}/Items/${id}?api_key=${API_KEY}"
> done
> ```
> 
> You can just throw this into a cronjob every night, this forces jellyfin to redownload pictures for the actors without one.


Hey man. Thanks for this. I implemented this in Unraid user scripts. After I run this, I think it fills up 97% of the missing actors/actresses. Just want to ask if I run this again, will it look only at those missing actors, or will it replace the current one?

**Comment by nosebeggar** [comment]
> > For anyone who happens to stumble upon this (It's been more than 2 years now, but the issue still persists), here is the linux version:
> > ```
> > #!/bin/bash
> > 
> > API_KEY="ajksjdijaskdjaiksjdkajsd"
> > BASE_URL="https://jellyfin.example.com"
> > USER="theUsernameYouLogInWith"
> > #Go to https://jellyfin.example.com/web/#/dashboard/users and "rightclick > copy link" on your portrait to get your user ID
> > USERID="31fd53576d384b288d055549dad87a4a"
> > 
> > # Fetch all persons
> > response=$(curl -s "${BASE_URL}/emby/Persons?api_key=${API_KEY}")
> > 
> > # Filter persons with empty ImageTags, sort by Id, and remove duplicates
> > ids=$(echo "$response" | jq -r '.Items[] | select(.ImageTags == null or .ImageTags == {}) | .Id' | sort -u)
> > 
> > # Fetch details for each person
> > for id in $ids; do
> >     curl -s "${BASE_URL}/Users/${USERID}/Items/${id}?api_key=${API_KEY}"
> > done
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > You can just throw this into a cronjob every night, this forces jellyfin to redownload pictures for the actors without one.
> 
> Hey man. Thanks for this. I implemented this in Unraid user scripts. After I run this, I think it fills up 97% of the missing actors/actresses. Just want to ask if I run this again, will it look only at those missing actors, or will it replace the current one?

It's documented in the script ;)
# Filter persons with empty ImageTags, sort by Id, and remove duplicates

Fires only against people with missing images

**Comment by One-Hoopy-Frood** [comment]
Confirmed having this issue on Jellyfin 10.10.7 on Ubuntu 24LTS.

Additional behavior:
I presume this may be the same as "clicking the actor", but if I click the [...] on bottom right of missing actor's icon, and hit "refresh metadata" >> (any option) it will also then refresh only that actor of the series.

Before...
<img width="1225" height="901" alt="Image" src="https://github.com/user-attachments/assets/15182f90-2223-4141-9b5b-25918a5d7a25" />

After...
<img width="961" height="837" alt="Image" src="https://github.com/user-attachments/assets/17cb552e-854a-4580-809a-a9d001c8651a" />

Will try the script.

**Comment by nosebeggar** [comment]
> Confirmed having this issue on Jellyfin 10.10.7 on Ubuntu 24LTS.
> 
> Additional behavior: I presume this may be the same as "clicking the actor", but if I click the [...] on bottom right of missing actor's icon, and hit "refresh metadata" >> (any option) it will also then refresh only that actor of the series.
> 
> Before... <img alt="Image" width="1225" height="901" src="https://private-user-images.githubusercontent.com/97187087/481633424-15182f90-2223-4141-9b5b-25918a5d7a25.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTYxMjg1MjgsIm5iZiI6MTc1NjEyODIyOCwicGF0aCI6Ii85NzE4NzA4Ny80ODE2MzM0MjQtMTUxODJmOTAtMjIyMy00MTQxLTliNWItMjU5MThhNWQ3YTI1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODI1VDEzMjM0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc4MzE4ZWViMDBlMGYyMGEwODVhNzgzZmM3YWEwMTcxNGQzM2NmMjBmNDVmNTQ5MDg0MmMwYWNmOTAzYTE3ZDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.QzVMEIrzx5dDvqeJkpWFoxcb8AFCgwt0tZhthvGBIH4">
> 
> After... <img alt="Image" width="961" height="837" src="https://private-user-images.githubusercontent.com/97187087/481633955-17cb552e-854a-4580-809a-a9d001c8651a.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTYxMjg1MjgsIm5iZiI6MTc1NjEyODIyOCwicGF0aCI6Ii85NzE4NzA4Ny80ODE2MzM5NTUtMTdjYjU1MmUtODU0YS00NTgwLTgwOWEtYTlkMDAxYzg2NTFhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODI1VDEzMjM0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk5YzgyMDhjNGM1ZGQyOTZjMzk2Y2JiNzNhYTU2ODQxODIyMWMyYjM1ZWY3N2YwZjA2NTY5ODBiNjJmM2VlN2QmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.WTZMx5d0_7_skpt4RzgVPr6Y2uEtz27UG3Re3Tc1qxE">
> 
> Will try the script.

Without actually having checked the sourcecode, i presume clicking the actor portrait just calls the same function if there is no picture, so this is the same thing, yes.

**Comment by LessTaken** [comment]
> # Filter persons with empty ImageTags, sort by Id, and remove duplicates
> ids=$(echo "$response" | jq -r '.Items[] | select(.ImageTags == null or .ImageTags == {}) | .Id' | sort -u)

First of all, thank you so much, I'm struggeling with fixing this issue now for a few months lol

But my Bash shows following error:
line 13: jq: command not found

Any idea why? I'm running Debian 12.

**Comment by j-insan3** [comment]
> > # Filter persons with empty ImageTags, sort by Id, and remove duplicates
> > ids=$(echo "$response" | jq -r '.Items[] | select(.ImageTags == null or .ImageTags == {}) | .Id' | sort -u)
> 
> First of all, thank you so much, I'm struggeling with fixing this issue now for a few months lol
> 
> But my Bash shows following error: line 13: jq: command not found
> 
> Any idea why? I'm running Debian 12.

Install jq (apt install jq)

**Comment by LessTaken** [comment]
> > > # Filter persons with empty ImageTags, sort by Id, and remove duplicates
> > > ids=$(echo "$response" | jq -r '.Items[] | select(.ImageTags == null or .ImageTags == {}) | .Id' | sort -u)
> > 
> > 
> > First of all, thank you so much, I'm struggeling with fixing this issue now for a few months lol
> > But my Bash shows following error: line 13: jq: command not found
> > Any idea why? I'm running Debian 12.
> 
> Install jq (apt install jq)

Insane fast reply lol
I'm not fully awaken yet, sorry for the dumb question tho. Found the missing package myself.
But thanks for the confirmation!

**Comment by HomelabHaven** [comment]
Still an issue with Jellyfin 10.11.1 (linuxserver.io docker-jellyfin image). I am running the script now to see if it helps.
Any update on when this issue might be fixed?

**Comment by cyanide699** [comment]
Same here issues that cast & crew don't refresh or auto populate
(Scheduled task to refresh people does appear complete successful but it doesn't work)

If I click on any of the placeholder icons the image then downloads 

Thanks to the script provided above this works fine.. (But really This should just work in jellyfin!)

Once I had raN through the script and all downloaded ok... I then ran a "refresh people" scheduled task...

This then broke it again and all images gone left with a black placeholder icon!.

So again I'm having to run the script and I have disabled the jellyfin "refresh people" task.

10.11.3 docker

**Comment by diznavis** [comment]
If the script isn't working, make sure you use a user id that has full rights on the server. I ran it once and it did nothing, but I had used a restricted user, ran it again using my admin account's user id, and it actually working.

Still would be nice to fix this without a script though, its still an issue on 10.11.5 installed on debian trixie (no docker, gave up on docker because it was unstable and extremely slow, running the install on the same hardware fixed it entirely).

**Comment by ghost** [comment]
same stupid bug in 2026...and people ask why we switch to emby........ 🤣🤣🤣🤣🤣🤣🤣

**Comment by ghost** [comment]
Hidden my comment as a span ....ahahahahahahhahaah ridiculous

**Comment by LessTaken** [comment]
*spam
And yes, it is actually spam since you just want to rant about this issue and not contributing at all

**Comment by HomelabHaven** [comment]
While I do think that @FotoFotoCheck's comment was not constructive, he does have a valid point and the frustration about something so simple still being an issue after so much time has passed is valid.

**Comment by geebru** [comment]
> While I do think that [@FotoFotoCheck](https://github.com/FotoFotoCheck)'s comment was not constructive, he does have a valid point and the frustration about something so simple still being an issue after so much time has passed is valid.

As an open source project, anyone that believes they know the fix is welcome to pick it up and fix it and have the team review it, I'm sure.

**Comment by ghost** [comment]
"I'm sure." ...i give to coder much solution to many bug...no one are implement..every one say "i'm superior you're nothing" ...ahahah Jellyfin is coded by child who think are able to write code... btw..have a great life with this "software" xD

**Comment by crplcreek** [comment]
I was having the same issue and noticed I had metadata locked for most of my content. All of movies and shows locked in mass using the metadata manager. Once I removed the lock and refreshed --can't recall if I scanned for missing or had to replace all and toggle to include replacing images --but the problem was solved.

Version 10.11.4 Docker image from TrueNAS SCALE apps catalog

**Comment by kuzyn** [comment]
I'm seeing this issue too. I think it appeared during the db migration update? Why does the "Refresh People" task do not fix this? Could be linked to #15649 ?

What I am seeing in the log
```
[2026-01-23 15:17:21.410 +00:00] [ERR] Error processing request: "Could not find file '/var/lib/jellyfin/metadata/People/A/Aden Young/folder.jpg'". URL "GET" "/Items/689794cebab37705835c1b64a592103d/Images/Primary".
[2026-01-23 15:17:21.412 +00:00] [ERR] Error processing request: "Could not find file '/var/lib/jellyfin/metadata/People/K/Kathleen Munroe/folder.jpg'". URL "GET" "/Items/45ed590bb63098dd216518c750121a90/Images/Primary".
[2026-01-23 15:17:21.414 +00:00] [ERR] Error processing request: "Could not find file '/var/lib/jellyfin/metadata/People/K/Karen Robinson/folder.jpg'". URL "GET" "/Items/7fe5edeeb0aaf047d2d5f7d790c45a0a/Images/Primary".
[2026-01-23 15:17:21.414 +00:00] [ERR] Error processing request: "Could not find file '/var/lib/jellyfin/metadata/People/K/K.C. Collins/folder.jpg'". URL "GET" "/Items/d8dd16dca4a1d9aa581d403732cf0baf/Images/Primary".
[2026-01-23 15:17:21.440 +00:00] [ERR] Error processing request: "Could not find file '/var/lib/jellyfin/metadata/People/D/Dick Wolf/folder.jpg'". URL "GET" "/Items/a3d3836b0f5e3f312a25b60b1c6e3208/Images/Primary".
[2026-01-23 15:17:21.517 +00:00] [ERR] Error processing request: "Could not find file '/var/lib/jellyfin/metadata/People/T/Tassie Cameron/folder.jpg'". URL "GET" "/Items/c2e63e9cb225f54a5990b48977261a56/Images/Primary".
```

**Comment by kiwichick888** [comment]
This is still an issue. Server version 10.10.6. What I don't understand is how it displays some cast/crew/guest star photos straight away but others not until clicked. It makes no sense.

**Comment by MDeveloping** [comment]
Also exist on Jellyfin 10.11.6. Hope for a fix in upcoming version.

**Comment by funkyfunked** [comment]
I just encountered this exact issue after 10.11.7 update coming from 10.11.6. Running "Refresh people" scheduled task does not resolve the issue.
